src/org/freenetproject/freemail/MailMessage.java

```diff
@@ -657,7 +657,7 @@ private static Charset extractCharsetFromContentType(MailMessage msg) throws Uns
 			String[] charsetParts = parts[1].trim().split("=", 2);
 			if(!charsetParts[0].equalsIgnoreCase("charset")) {
 				throw new UnsupportedEncodingException("Can't handle text/plain with parameter other than charset. "
-						+ "Unrecognised parameters were: " + params);
+						+ "Unrecognised parameters were: " + charsetParts);
```

Comment:
That one almost slipped past, it needs to be:
```
"Unrecognised parameters were: " + toString(charsetParts)
```
with a static import of `Arrays.toString`.